DS13/code/modules/EMH/EMH.dm

```diff
@@ -0,0 +1,291 @@
+
+/obj/machinery/emh_emitter/proc/EMH_present()
+	var/mob/living/carbon/human/species/holographic/S = locate(/mob/living/carbon/human/species/holographic) in GLOB.alive_mob_list
+	if(istype(S) && !QDELETED(S))
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/emh_emitter/examine(mob/user)
+	. = ..()
+	if(EMH_present())
+		if(emh.client)
+			to_chat(user, "<i>The panel shows that there is an active EMH on the network.</i>")
+		else
+			to_chat(user, "<i>The panel shows that there is an EMH on the network, but it is not responding (braindead). Click the panel to summon a new one.</i>")
+
+/mob/verb/editemhblacklist()
+	set name = "Edit EMH blacklist"
+	set desc = "Remove people from the EMH blacklist for this round."
+	set category = "Admin"
+	if(!check_rights(R_ADMIN, TRUE)) //We may want to change this to R_VAREDIT when we launch but for now I'd quite like mentors to have this too.
+		return
+	var/A
+	var/list/temp = GLOB.EMH_blacklist.Copy()
+	temp += "cancel" //To stop the input only having one option, which makes byond freak out.
+	A = input("Remove whom from the blacklist?", "Admin tools", A) as null|anything in GLOB.EMH_blacklist
+	if(!A)
+		return
+	if(A in GLOB.EMH_blacklist)
+		GLOB.EMH_blacklist -= A
+	log_admin("([worldtime2text()]):[src] / [client.ckey] removed [A] from the EMH blacklist for this round.")
+
+/obj/machinery/emh_emitter/attackby(obj/I,mob/user)
+	. = ..()
+	if(user.client)
+		var/obj/item/card/id/ID = I
+		if(ID && istype(ID))
+			if(check_access(ID))
+				var/question = alert("Terminate the current EMH? (this kills the current EMH)",name,"yes","no")
+				var/blacklist = alert("Blacklist the current EMH player for the duration of this round? ",name,"yes","no")
+				if(question == "yes")
+					if(!emh)
+						emh = locate(/mob/living/carbon/human/species/holographic) in GLOB.alive_mob_list
+					if(emh.client && emh.client.key && blacklist == "yes")
+						GLOB.EMH_blacklist += emh.client.key
+						to_chat(emh, "<span_class='boldnotice'>[user] has blacklisted you from becoming an EMH for the duration of this round! If you feel this was in error, please ahelp immediately.")
+						log_game("[user] just blacklisted [emh.client.key] from becoming an EMH for the rest of the round")
+					if(emh)
+						emh.death()//Kill the emh.
+						emh = null
+					else
+						to_chat(user, "Unable to locate an EMH on the network.")
+
+/obj/machinery/emh_emitter/Destroy()
+	if(emh)
+		var/obj/machinery/emh_emitter/saveme = locate(/obj/machinery/emh_emitter) in GLOB.machines
+		if(saveme)
+			to_chat(emh, "The emitter you were using has been destroyed, your program has been transferred to avoid corruption.")
+			saveme.activate(emh)
+		else
+			to_chat(emh, "The emitter you were using was destroyed! Your program could not be shunted to another emitter and will be shut down to avoid damage.")
+			qdel(emh)
+			emh = null
+	STOP_PROCESSING(SSmachines,src)
+	. = ..()
+
+/obj/machinery/emh_emitter/attack_hand(mob/user)
```

Comment:
This needs a limiter to prevent people hitting it constantly to keep causing the popup for the EMH constantly. Or some sort of other check to prevent a request to transfer to same location that the EMH is at.